client/src/components/Donation/PaypalButton.tsx

```diff
@@ -37,7 +85,10 @@ export class PaypalButton extends Component {
     return { ...configurationObj };
```

Comment:
Hey @ahmadabdolsaheb could you take a look at the (state === configurationObj) conditional above?  @Nrestrepo05 noticed some issues with it and that made me realise that it's never `true`, since that's only possible if `configurationObj` is literally the `state` object.

Any clue if this matters?  Did we need it to return null in some cases?

Reply:
I see, that was supposed to prevent the state from changing when props change. 
it should not matter much since `donationAmount, donationDuration` props don't change ( we have one subscription plan)

Reply:
I see, thanks. In that case I recommend removing that comparison and leaving a comment to re-implement it as a deep comparison.

Reply:
Thanks, I'll do it!